File: unityPart/Assets/Scripts/Lua/Managers/SpawnManager.lua
local Vector3 = CS.UnityEngine.Vector3
local Quaternion = CS.UnityEngine.Quaternion

local export = {}
local UIMap = {}

function export.SpawnPrefab(name, position, rotation)
    local _SpawnManager = CS.Base.SpawnManager
    return _SpawnManager.SpawnPrefab(name, position, rotation)
end

-- function export.CloseUI(name)
--     print("CloseUI：" .. name)
--     _UIManager.CloseUI(name);
-- end

return export

File: unityPart/Assets/Scripts/Managers/SpawnManager.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using UnityEngine.AddressableAssets;
using UnityEngine.ResourceManagement.AsyncOperations;
using System.Threading.Tasks;
using XLua;

namespace Base
{
    [LuaCallCSharp]
    public class SpawnManager : IManager
    {
        public static Dictionary<string, GameObject> prefabsMap = new Dictionary<string, GameObject>();
        public string[] prefabNames = { 
            "Assets/Prefabs/Roler/BlackMan.prefab",
            "Assets/Prefabs/Roler/NewBlackMan.prefab"
        };
        public void Awake()
        {
            LoadAllPrefab();
        }

        private async void LoadAllPrefab()
        {
            for(int i = 0; i < prefabNames.Length; i++)
            {
                GameObject prefab = await Addressables.LoadAssetAsync<GameObject>(prefabNames[i]).Task;
                prefabsMap.Add(prefabNames[i], prefab);
            }
        }

        public static GameObject SpawnPrefab(string name, Vector3 position, Quaternion rotation)
        {
            GameObject gameObject = GameObject.Instantiate(prefabsMap[name], position, rotation);
            return gameObject;
        }

        public void OnloadOver()
        {

        }

        public void Update()
        {

        }

        public void Start()
        {

        }
    }
}

